### PR TITLE
Re-committing changes for mosaic tracking update after faulty merge.

### PR DESCRIPTION
**Pull request by SgStapleton**
Ran into an issue attempting to push last changes up to PR https://github.com/USGS-Astrogeology/ISIS3/pull/385. This current PR should now contain the final product for the major update of how tracking is being handled now for the various mosaicking applications.

### CODE REVIEW

File: isis/src/base/apps/automos/automos.xml [hunk 1]
```diff
@@ -61,19 +61,41 @@
...
+      </p>
+      <p>
+        The TRACK feature works with Priority options ONTOP and BENEATH for single band input
+        cubes. It works for multiband cubes for PRIORITY=ONTOP only when the NULL,
+        HIGHSATURATION and LOWSATURATION options are set to true. It also works for multiband
+        cubes when PRIORITY=BAND. Furthermore, this feature is NOT supported when
+        PRIORITY=AVERAGE.
+      </p>
+      <p>
+        <b>Please Note: Older versions of automos (prior to isis3.5.3) no longer work with
```
Comment (jessemapel) [inline isis/src/base/apps/automos/automos.xml]: We're releasing 3.6.0 so say versions before 3.6.0

File: isis/src/base/apps/handmos/handmos.xml [hunk 2]
```diff
@@ -47,19 +47,41 @@
...
+    </p>
+    <p>
+      The TRACK feature works with Priority options ONTOP and BENEATH for single band input
+      cubes. It works for multiband cubes for PRIORITY=ONTOP only when the NULL,
+      HIGHSATURATION and LOWSATURATION options are set to true. It also works for multiband
+      cubes when PRIORITY=BAND. Furthermore, this feature is NOT supported when
+      PRIORITY=AVERAGE.
+    </p>
+    <p>
+      <b>Please Note: Older versions of handmos (prior to isis3.5.3) no longer work with
```
Comment (jessemapel) [inline isis/src/base/apps/handmos/handmos.xml]: The next release is 3.6.0 so this should be `(prior to 3.6.0)`

File: isis/src/base/apps/handmos/handmos.xml [hunk 3]
```diff
@@ -47,19 +47,41 @@
...
+      mosaics that contain an attached tracking band. Mosaics including an attached tracking
+      cube can be updated and reformatted by running the program trackextract. This new
+      utility will create a new version of the mosaic with the appropriate label information
+      and a separate tracking cube to which it points.</b>
```
Comment (jessemapel) [inline isis/src/base/apps/handmos/handmos.xml]: this paragraph doesn't make sense to me. It seems like this has attached tracking cube and attached tracking band mixed up. The tracking cube is the new way, the tracking band is the old way. Also is this talking about the fact that old versions of handmos don't see the tracking cube or that new versions of handmos don't work with the tracking band? Both are compatibility issues, but it seems like they are getting mixed up here.

File: isis/src/base/apps/handmos/handmos.xml [hunk 4]
```diff
@@ -436,6 +467,11 @@
       group in the cube labels which could impact processing scripts/pipelines. Fixes #1620.
       References #1623.
     </change>
+    <change name="Summer Stapleton" date="2018-08-13">
+      Updated code and documentation to reflect new handling of tracking capabilities with an 
+      external tracking cube as well as clarify why special pixel flags are required when 
+      priority=ontop for multiband mosaics. References #2092
+    </change>
```
Comment (jessemapel) [inline isis/src/base/apps/handmos/handmos.xml:476]: I really like how clear the special pixel documentation is

File: isis/src/base/apps/mapmos/mapmos.xml [hunk 5]
```diff
@@ -59,19 +59,41 @@
...
+        HIGHSATURATION and LOWSATURATION options are set to true. It also works for multiband
+        cubes when PRIORITY=BAND. Furthermore, this feature is NOT supported when
+        PRIORITY=AVERAGE.
+      </p>
+      <p>
+        <b>Please Note: Older versions of mapmos (prior to isis3.5.3) no longer work with
+        mosaics that contain an attached tracking band. Mosaics including an attached tracking
+        cube can be updated and reformatted by running the program trackextract. This new
+        utility will create an new version of the mosaic with the appropriate label information
+        and a separate tracking cube to which it points. </b>
```
Comment (jessemapel) [inline isis/src/base/apps/mapmos/mapmos.xml]: See my handmos comments

File: isis/src/base/apps/trackextract/trackextract.xml [hunk 6]
```diff
@@ -6,14 +6,21 @@
...
-    date with isis version 3.5.3 or above. Removes the tracking band and table from the input mosaic 
-    cube and places them into a separate tracking cube. A reference to the tracking cube is added to 
-    the mosaic label. The mosaic cube's bit type remains the same, but the tracking cube's bit type 
-    is made unsigned integer. We also have to update the tracking cube's pixels by subtracting the 
-    offsett ProcessMosaic adds to each pixel and adding the new offset, the minimum value of an 
-    unsigned integer. ProcessMosaic also sets pixels that are not from any cube to a default value. 
-    If a pixel has this value, we set it to Null.
+    <p>
+      Updates a mosaic cube with tracking information that was created using older versions of ISIS
+      (prior to 3.5.3) to be up to date with ISIS version 3.5.3 and above. Removes the tracking
```
Comment (jessemapel) [inline isis/src/base/apps/trackextract/trackextract.xml]: 3.6.0 is the next release, not 3.5.3

File: isis/src/base/apps/trackextract/trackextract.xml [hunk 7]
```diff
@@ -6,14 +6,21 @@
...
+      band and the InputImages table from the input mosaic cube and places them into a separate
+      tracking cube. A reference to the tracking cube is then added to the mosaic label. In the
+      copying process, if a pixel in the tracking band of the input mosaic is not associated with
+      any of the original input images, a NULL value is written to that pixel in the tracking cube.
+      The mosaic cube's bit type remains the same, but the tracking cube's bit type will always be
+      unsigned integer. 
+    </p>
+    <p>
+      <b>Please Note: This application only works for 32-bit ("Real") mosaics as both 8-bit and
+      16-bit mosaics created prior to ISIS version 3.5.3 are considered to be corrupted.</b>
```
Comment (jessemapel) [inline isis/src/base/apps/trackextract/trackextract.xml]: Another version number

File: isis/src/base/apps/trackextract/trackextract.xml [hunk 8]
```diff
@@ -35,8 +48,10 @@
           Input cube with tracking band
         </brief>
         <description>
-          The mosaic created on or before the isis version 3.5.2 that contains the tracking band to 
-          be extracted.
+          The mosaic created on or before the ISIS version 3.5.2 that contains the tracking band
+          and InputImages table to be extracted. <b>Please Note: This mosaic cube must be of
+          bit-type "Real" as both 8-bit and 16-bit mosaics created before ISIS version 3.5.3 are
```
Comment (jessemapel) [inline isis/src/base/apps/trackextract/trackextract.xml]: 3.6.0

File: isis/src/base/apps/trackextract/trackextract.xml [hunk 9]
```diff
@@ -50,9 +65,10 @@
           Output mosaic and tracking cube
         </brief>
         <description>
-          The output mosaic cube name that will be up to date with isis version 3.5.3 or above. 
-          This cube will not have the tracking band, but instead the tracking information will be 
-          placed into a new cube with the same name but with "_tracking" added to the end.
+          The output mosaic cube name that will be up to date with isis version 3.5.3 and above.
```
Comment (jessemapel) [inline isis/src/base/apps/trackextract/trackextract.xml]: 3.6.0

File: isis/src/base/objs/TrackingTable/TrackingTable.cpp [hunk 10]
```diff
@@ -22,6 +22,7 @@
 
 #include "TrackingTable.h"
 
+#include <QDebug>
```
Comment (jessemapel) [inline isis/src/base/objs/TrackingTable/TrackingTable.cpp]: This include can be removed

**Review by jessemapel — CHANGES_REQUESTED** [review]
Looks good overall, just has a few documentation tweaks needed

File: isis/src/base/apps/handmos/handmos.xml (hunk 4 above)
Comment (SgStapleton) [reply isis/src/base/apps/handmos/handmos.xml:476]: Thank you. :-)

**Comment by SgStapleton** [comment]
@jessemapel Does this explanation make a little more sense...? (The "Please Note" block starting at line 80 in handmos.xml and repeated in the other documentations.)

**Review by jessemapel — APPROVED** [review]

**Comment by jessemapel** [comment]
Looks good

**Pull request merged**